cli/args/mod.rs

```diff
@@ -32,26 +36,36 @@ use deno_core::anyhow::Context;
 use deno_core::error::AnyError;
 use deno_core::normalize_path;
 use deno_core::parking_lot::Mutex;
+use deno_core::serde_json;
 use deno_core::url::Url;
+use deno_graph::npm::NpmPackageReq;
 use deno_runtime::colors;
+use deno_runtime::deno_node::PackageJson;
 use deno_runtime::deno_tls::rustls;
 use deno_runtime::deno_tls::rustls::RootCertStore;
 use deno_runtime::deno_tls::rustls_native_certs::load_native_certs;
 use deno_runtime::deno_tls::rustls_pemfile;
 use deno_runtime::deno_tls::webpki_roots;
 use deno_runtime::inspector_server::InspectorServer;
 use deno_runtime::permissions::PermissionsOptions;
+use once_cell::sync::Lazy;
 use std::collections::BTreeMap;
+use std::collections::HashMap;
+use std::collections::HashSet;
 use std::env;
 use std::io::BufReader;
 use std::io::Cursor;
 use std::net::SocketAddr;
 use std::num::NonZeroUsize;
+use std::path::Path;
 use std::path::PathBuf;
 use std::sync::Arc;
 
 use crate::cache::DenoDir;
 use crate::file_fetcher::FileFetcher;
+// TODO(bartlomieju): refactor this bit; we have a circular dependency
+// between this file and `cli/npm/resolver/mod.rs`.
+use crate::npm::NpmProcessState;
```

Comment:
Is it hard to move this to into this args folder?

Reply:
Depends on `NpmResolutionSnapshot` which is in `cli/npm/resolution.rs`. Maybe let's clean this up in a follow up?

Reply:
I think actually it might be ok for `NpmProcessState` to live in the npm module. I see it's used to get the current state to pass down to the process.